### PR TITLE
Modify S3431: Promote C# rule to SonarWay

### PR DESCRIPTION
**Pull request by sebastien-marichal**
Update to LaYC format

### CODE REVIEW

File: rules/S3431/exceptions.adoc [hunk 1]
```diff
@@ -1,3 +1,5 @@
 === Exceptions
 
-This rule ignores one-line test methods, since it is obvious in such methods where the exception is expected to be thrown. 
\ No newline at end of file
+This rule ignores:
+* single-line tests, since it is obvious in such methods where the exception is expected to be thrown
+* tests when it tests control flow and assertion are present in either a `catch` or `finally` clause
```
Comment (zsolt-kolbay-sonarsource) [inline rules/S3431/exceptions.adoc:5]: It might be worth adding a code example here.

File: rules/S3431/csharp/how-mstest.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
+== How to fix it in MSTest
+
+Remove the `ExpectedException` attribute in favor of using the https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.assert.throwsexception[Assert.ThrowsException] assertion.
+
+=== Code examples
+
+==== Noncompliant code example
+
+[source,csharp,diff-id=2,diff-type=noncompliant]
```
Comment (zsolt-kolbay-sonarsource) [inline rules/S3431/csharp/how-mstest.adoc:9]: Nitpick: this should be `diff-id=1`, because it's included first in the rule.adoc file.

**Review by zsolt-kolbay-sonarsource — APPROVED** [review]
LGTM

**Comment by sebastien-marichal** [comment]
@zsolt-kolbay-sonarsource I am asking you to review again; I made more changes than what you suggested.

**Review by zsolt-kolbay-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**